src/execution/operator/persistent/physical_insert.cpp

```diff
@@ -343,9 +363,85 @@ idx_t PhysicalInsert::OnConflictHandling(TableCatalogEntry &table, ExecutionCont
 		}
 	}
 
-	RegisterUpdatedRows(lstate, row_ids, combined_chunk.size());
+	RegisterUpdatedRows(lstate, row_ids, combined_chunk.size(), true);
+
+	idx_t updated_tuples = PerformGlobalOnConflictAction(context, combined_chunk, table, row_ids);
+
+	// Remove the conflicting tuples from the insert chunk
+	SelectionVector sel_vec(lstate.insert_chunk.size());
+	idx_t new_size =
+	    SelectionVector::Inverted(conflicts.Selection(), sel_vec, conflicts.Count(), lstate.insert_chunk.size());
+	lstate.insert_chunk.Slice(sel_vec, new_size);
+	lstate.insert_chunk.SetCardinality(new_size);
+	return updated_tuples;
+}
+
+idx_t PhysicalInsert::HandleLocalConflicts(TableCatalogEntry &table, ExecutionContext &context,
```

Comment:
It seems like there is a lot of duplicate code in `HandleGlobalConflicts` and `HandleLocalConflicts`. Is it possible to unify that a bit more?